server/src/main/java/org/opensearch/rest/action/cat/RestIndicesAction.java

```diff
@@ -205,6 +235,20 @@ public void onFailure(final Exception e) {
         };
     }
 
+    private void validateRequestLimit(final ClusterStateResponse clusterStateResponse, final ActionListener<Table> listener) {
+        if (isRequestLimitCheckSupported() && Objects.nonNull(clusterStateResponse) && Objects.nonNull(clusterStateResponse.getState())) {
+            int limit = requestLimitSettings.getCatIndicesLimit();
+            if (RequestLimitSettings.isCircuitLimitBreached(clusterStateResponse.getState().getMetadata(), INDICES, limit)) {
```

Comment:
Can we verify if the http response is 429 when the limit is breached ?

Reply:
Yes, its 429